### ISSUE TITLE
Pytorch 1.5+ is slower than pytorch 1.3

### ISSUE THREAD

**Issue by elepherai** [pytorch/pytorch#61243]
Hi, I use solov2 for instance segmentation and I found that the inference speed of using pytorch 1.5+ is about 3 times slower than pytorch 1.3.

The codes are almost the same except changing AT_CHECK to TORCH_CHECK in pytorch 1.5+.

The base docker images are pytorch/pytorch:1.3-cuda10.1-cudnn7-devel , pytorch/pytorch:1.9.0-cuda11.1-cudnn8-devel and 1.6.0-cuda10.1-cudnn7-devel.

**Comment by xsacha** [comment]
What GPU are you testing on? It's possible the CUDA code is no longer pre-made and relies on PTX now if it's an older arch.

**Comment by elepherai** [comment]
@xsacha 
Hi, I test on TITAN Xp and set TORCH_CUDA_ARCH_LIST="6.0 6.1 7.0+PTX".

**Comment by xsacha** [comment]
You compiled these PyTorch (with the TORCH_CUDA_ARCH_LIST) or they are part of the base docker?

**Comment by elepherai** [comment]
@xsacha 
Yes, they are part of the base docker, [Dockerfile](https://github.com/WXinlong/SOLO/blob/master/docker/Dockerfile).

**Comment by xsacha** [comment]
It's possibly an issue with CUDNN 7->8. Are you able to try it with CUDNN 7 instead?

**Comment by elepherai** [comment]
I have tried cudnn7 but unfortunately got slower speed. 
 
> The base docker images are pytorch/pytorch:1.3-cuda10.1-cudnn7-devel , pytorch/pytorch:1.9.0-cuda11.1-cudnn8-devel and 1.6.0-cuda10.1-cudnn7-devel.

**Comment by malfet** [comment]
Can you provide a bit more details how you benchmark inference speed? Are you simply timing https://github.com/WXinlong/SOLO/blob/master/demo/inference_demo.py ?

**Comment by elepherai** [comment]
@malfet Hi, I tested inference_demo.py and found that the problem is not about different pytorch versions but my own inference code.

But I'm still confused about the difference of inference speed between my code and the demo code.

My intention is to speedup pre-processing by converting the normalization from cpu to gpu. I got 3 ~ 4 times faster in pre-processing but 1.4 ~ 2 times slower in model inference. 

Same input tensor to model inference, same model inference code and same prediction result but difference inference speed.
inference_demo_0
![image](https://user-images.githubusercontent.com/43477405/126958888-b5a408d6-b398-4dd9-925f-41f73359cfc6.png)
inference_demo_1
![image](https://user-images.githubusercontent.com/43477405/126958911-a55ec733-11d4-4e59-ac96-cdc7807dbd83.png)



Codes:
[inference_demo_0.py](https://github.com/pytorch/pytorch/files/6876703/inference_demo_0.py.txt)
[inference_demo_1.py](https://github.com/pytorch/pytorch/files/6876705/inference_demo_1.py.txt)

Image:
[test.jpg](https://user-images.githubusercontent.com/43477405/126957681-2a44fa67-b117-4613-a3ba-3650728757b6.jpg)

Model:
[weights](https://cloudstor.aarnet.edu.au/plus/s/HwHys05haPvNyAY/download)
[config](https://github.com/WXinlong/SOLO/blob/master/configs/solov2/solov2_light_448_r18_fpn_8gpu_3x.py)

Thanks~
